qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+            self.resource_group_name,
+            self.storage_account,
+            self.session,
+        )
+
+        job_id = container[0]
+        container_name = container[1]
+
+        job = input_run
+        routes = self._helpers.create_job_routes(
+            self.session,
+            container_name,
+            job,
+            self.connection_string,
+            self.location_name,
+            self.subscription_id,
+            self.resource_group_name,
+            self.workspace_name,
+            quantum_access_token,
+        )
+
+        containerSasUri = routes[0]
+        inputSasUri = routes[1]
+        outputSasUri = routes[2]
+        job_headers = routes[3]
+
+        job_payload = self._helpers.create_payload(
+            containerSasUri,
+            inputSasUri,
+            outputSasUri,
+            job_id,
+            job_name=name,
+            input_data_format=self.formats[provider][0],
+            provider=provider,
+            backend=self.backends[backend],
+            num_qubits=qubits,
+            num_shots=100,
+            total_count=100,
+            output_data_format=self.formats[provider][1],
+        )
+
+        job_approval = self._helpers.submit_job(
+            job_id,
+            job_payload,
+            job_headers,
+            self.location_name,
+            self.subscription_id,
+            self.resource_group_name,
+            self.workspace_name,
+            self.session,
+        )
+
+        self.jobs[job_id] = [job_headers, container_name]
+
+        return job_approval
+
+    def get_job(self, job_id: str) -> dict[str, Any]:
+        """Get a specific Azure Quantum job."""
+
+        job = job_id
+        header = self.jobs[job_id][0]
+        url = (
+            f"https://{self.location_name}.quantum.azure.com/subscriptions/{self.subscription_id}"
+            f"/resourceGroups/{self.resource_group_name}/providers/Microsoft.Quantum/workspaces/"
+            f"{self.workspace_name}/jobs/{job}?api-version=2022-09-12-preview"
+        )
+
+        response = self.session.get(url, headers=header)
+        return response.json()
+
+    def get_job_data(self, job_id: str) -> dict[str, Any]:
+        """Get the data from a specific Azure Quantum job."""
+
+        data = self._helpers.get_output_data(f"{self.jobs[job_id][1]}", self.connection_string)
+        return data
+
+    def cancel_job(self, job_id: str):
+        """Cancel a specific Azure Quantum job."""
+        job = job_id
+        header = self.jobs[job_id][0]
+        url = (
+            f"https://{self.location_name}.quantum.azure.com/subscriptions/"
+            f"{self.subscription_id}/resourceGroups/{self.resource_group_name}/"
+            f"providers/Microsoft.Quantum/workspaces/{self.workspace_name}/jobs/"
+            f"{job}?api-version=2022-09-12-preview"
+        )
+
+        response = self.session.delete(url, headers=header)
+        print("Job has been cancelled.")
```

Comment:
Use `logger.info` instead of print. Other developers who use the package can configure log-level to show which stdout / stderr messages they want to see. print bypasses that and prints to the console automatically without any easy way for people to shut it off